src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArraysTests.cs

```diff
@@ -0,0 +1,508 @@
+        {
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+
+public class A
+{
+    public void B()
+    {
+        Console.WriteLine(string.Join("" "", {|CA1854:new[] { ""a"", ""b"" }|} /* test comment */));
+    }
+}
+", @"
+using System;
+
+public class A
+{
+    private static readonly string[] value = new[] { ""a"", ""b"" };
+
+    public void B()
+    {
+        Console.WriteLine(string.Join("" "", value /* test comment */));
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task IdentifyConstArrays_LambdaArrayCreation()
+        {
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+using System.Linq;
+
+public class A
+{
+    public void B()
+    {
+        var x = new string[] { ""a"", ""b"" };
+        var y = x.Select(z => {|CA1854:new[] { ""c"" }|});
+    }
+}
+", @"
+using System;
+using System.Linq;
+
+public class A
+{
+    private static readonly string[] stringArray = new[] { ""c"" };
+
+    public void B()
+    {
+        var x = new string[] { ""a"", ""b"" };
+        var y = x.Select(z => stringArray);
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task IdentifyConstArrays_LambdaInvokedArrayCreation()
+        {
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+using System.Linq;
+
+public class A
+{
+    public void B()
+    {
+        var x = new string[] { ""a"", ""b"" };
+        var y = x.Select(z => {|CA1854:new[] { ""c"" }|}.First());
+    }
+}
+", @"
+using System;
+using System.Linq;
+
+public class A
+{
+    private static readonly string[] sourceArray = new[] { ""c"" };
+
+    public void B()
+    {
+        var x = new string[] { ""a"", ""b"" };
+        var y = x.Select(z => sourceArray.First());
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task IdentifyConstArrays_ExtensionMethod()
+        {
+            await VerifyCS.VerifyCodeFixAsync(@"
+using System;
+using System.Linq;
+
+public class A
+{
+    public void B()
+    {
+        string y = {|CA1854:new[] { ""a"", ""b"", ""c"" }|}.First();
+        Console.WriteLine(y);
+    }
+}
+", @"
+using System;
+using System.Linq;
+
+public class A
+{
+    private static readonly string[] sourceArray = new[] { ""a"", ""b"", ""c"" };
+
+    public void B()
+    {
+        string y = sourceArray.First();
+        Console.WriteLine(y);
+    }
+}
+");
+
+            await VerifyVB.VerifyCodeFixAsync(@"
+Imports System
+Imports System.Linq
+
+Public Class A
+    Public Sub B()
+        Dim y As String = {|CA1854:{""a"", ""b"", ""c""}|}.First()
+        Console.WriteLine(y)
+    End Sub
+End Class
+", @"
+Imports System
+Imports System.Linq
+
+Public Class A
+    Private Shared ReadOnly stringArray As String() = {""a"", ""b"", ""c""}
+    Public Sub B()
+        Dim y As String = stringArray.First()
+        Console.WriteLine(y)
+    End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public async Task IdentifyConstArrays_MemberExtractionTest()
```

Comment:
Don't know if it's better to update this test or add another one where there would be more than one conflict for the field name.

Reply:
That's already in the test.